### PR TITLE
XSH-835 Allow specifying custom headers in GoodPandas constructor

### PR DESCRIPTION
**Pull request by Dominik-Sauer**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-pandas/gooddata_pandas/good_pandas.py [hunk 1]
```diff
@@ -17,11 +17,23 @@ class GoodPandas:
...
-        self._sdk = GoodDataSdk.create(host, token, USER_AGENT, Host=headers_host)
+    def __init__(
+        self,
+        host: str,
+        token: str,
+        headers_host: Optional[str] = None,
+        **custom_headers_: Optional[str],
+    ) -> None:
+        if headers_host is not None:
+            custom_headers_["Host"] = headers_host
```
Comment (hkad98) [inline gooddata-pandas/gooddata_pandas/good_pandas.py:28]: IMHO this is not necessary. We can have Host=None. The GoodDataSdk filters out headers with None as a value, but if you want to keep it like this, we can.

File: gooddata-pandas/gooddata_pandas/good_pandas.py [hunk 2]
```diff
@@ -17,11 +17,23 @@ class GoodPandas:
...
+        self._sdk = GoodDataSdk.create(host, token, USER_AGENT, **custom_headers_)
         self._series_per_ws: dict[str, SeriesFactory] = dict()
         self._frames_per_ws: dict[str, DataFrameFactory] = dict()
 
+    @property
+    def sdk(self) -> GoodDataSdk:
+        return self._sdk
```
Comment (hkad98) [inline gooddata-pandas/gooddata_pandas/good_pandas.py:35]: Is there a use case why for exposing SDK? If not, I would add it when it is relevant.

File: gooddata-pandas/gooddata_pandas/good_pandas.py (hunk 1 above)
Comment (Dominik-Sauer) [reply gooddata-pandas/gooddata_pandas/good_pandas.py:28]: That's to retain backwards compatibility allowing use of headers_host variable. However I don't want to override the custom_headers_' "Host" entry when the headers_host is absent.

File: gooddata-pandas/gooddata_pandas/good_pandas.py (hunk 2 above)
Comment (Dominik-Sauer) [reply gooddata-pandas/gooddata_pandas/good_pandas.py:35]: In tabular-exporter we're accessing the _sdk, I want to fix it in follow-up change.

File: gooddata-pandas/gooddata_pandas/good_pandas.py (hunk 1 above)
Comment (hkad98) [reply gooddata-pandas/gooddata_pandas/good_pandas.py:28]: Fair enough. Resolving conversation.

File: gooddata-pandas/gooddata_pandas/good_pandas.py (hunk 2 above)
Comment (hkad98) [reply gooddata-pandas/gooddata_pandas/good_pandas.py:35]: Okay, this reason is enough. Closing conversation.

**Review by hkad98 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR160

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/146/ : SUCCESS in 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/146/ : FAILURE in 17s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/148/ : SUCCESS in 19s

**Pull request merged**